c7n/resources/workspaces.py

```diff
@@ -463,6 +463,41 @@ def process(self, directories):
             )
 
 
+class DescribeWorkspacesWeb(DescribeSource):
+
+    def augment(self, resources):
+        resources = universal_augment(self.manager, resources)
+
+        client = local_session(self.manager.session_factory).client('workspaces-web')
+        results = []
+
+        for r in resources:
+            try:
+                r['networkSettings'] = client.get_network_settings(
```

Comment:
so we generally want to structure these additional enrichments from separate api calls as dedicated filters such that a policy that only cares about one doesn't have to pay the additional time / api rate limit on the additional calls not germane to the policy.